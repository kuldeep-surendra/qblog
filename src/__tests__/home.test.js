import React from 'react';
import ReactDOM from 'react-dom';
import Home from '../components/home';
import { shallow, mount, render, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('Home', () => {
  it('should mount home component', () => {
    const wrapper = shallow(<Home />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should click create blog post link", () => {
    const wrapper = shallow(<Home />)
    const card = wrapper.find('Columns').find('Box').first().find('Card')
    const component = renderer.create(card.props().link)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("should click sample table link", () => {
    const wrapper = shallow(<Home />)
    const card = wrapper.find('Columns').find('Box').at(2).find('Card')
    const component = renderer.create(card.props().link)
    let tree = component.toJSON();
    tree.props.onClick();
    expect(tree).toMatchSnapshot();
  })

  it("should click posts feed link", () => {
    const wrapper = shallow(<Home />)
    const card = wrapper.find('Columns').find('Box').at(1).find('Card')
    const component = renderer.create(card.props().link)
    let tree = component.toJSON();
    tree.props.onClick();
    expect(tree).toMatchSnapshot();
  })
})
