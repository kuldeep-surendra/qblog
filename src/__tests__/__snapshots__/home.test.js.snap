// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should mount home component 1`] = `
<div>
  <Connect(NavBar) />
  <br />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    margin="large"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Columns
      justify="start"
      maxCount={1}
      responsive={true}
    >
      <Box
        announce={false}
        direction="column"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div"
      >
        <Card
          a11yTitle="Card"
          contentPad="medium"
          description="Create your blog and spread knowledge!"
          heading="Create Blog Post"
          headingStrong={true}
          label="Blog post"
          link={
            <Anchor
              animateIcon={true}
              label="Create >>"
              method="push"
              onClick={[Function]}
              tag="a"
            />
          }
          textSize="small"
        />
      </Box>
      <Box
        announce={false}
        direction="column"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div"
      >
        <Card
          a11yTitle="Card"
          contentPad="medium"
          description="Get details of all posts, explore knowledge."
          heading="Posts Feed"
          headingStrong={true}
          label="All Posts"
          link={
            <Anchor
              animateIcon={true}
              label="List >>"
              method="push"
              onClick={[Function]}
              tag="a"
            />
          }
          textSize="small"
        />
      </Box>
      <Box
        announce={false}
        direction="column"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div"
      >
        <Card
          a11yTitle="Card"
          contentPad="medium"
          description="Sample description providing more details."
          heading="Sample Heading"
          headingStrong={true}
          label="Sample Table"
          link={
            <Anchor
              animateIcon={true}
              href=""
              label="Sample Table"
              method="push"
              onClick={[Function]}
              tag="a"
            />
          }
          textSize="small"
        />
      </Box>
    </Columns>
  </Box>
</div>
`;
